### ISSUE TITLE
CUDA out of memery at start of DISTRIBUTED training and slow training speed

### ISSUE THREAD

**Issue by tianxinhang** [open-mmlab/mmdetection#3873]
dataset: 118k coco2017 train
model: htc-res50
load_from: None
CUDA:10.2
pytorch:1.5
python:3.7
mmdetection: v2.4 (just git from open_mmlab)

issue1: I tried to train htc res50 on 3 different mechines and they got 2 GPUs, 4 gpus, 8 gpus respectively. For examlple, on 2 GPUs, I ran the command ./tools/dist_train.sh configs/htc/htc_r50_fpn_1x_coco.py   2  --work-dir /home/work_dir_01/, the config log of gpus_ids should be range(0,2).
But:
the log is like this 
....
load_from = None
resume_from = None
workflow = [('train', 1)]
work_dir = '/home/work_dir_01/'
gpu_ids = range(0, 1).

AND, in another two mechines, when I ran  ./tools/dist_train.sh configs/htc/htc_r50_fpn_1x_coco.py   4  --work-dir /home/work_dir_01/   and   ./tools/dist_train.sh configs/htc/htc_r50_fpn_1x_coco.py   8  --work-dir /home/work_dir_01/ respectively, the gpu_ids is still range(0,1).  However, for 8 gpus mechines, this training is running and when I type nvidia-smi, the all the gpus are working! 

But,for 2 and 4 gpus mechines, the training stuck at start stage of trainging, it both shows up CUDA out of memery.Tried to allocate 61.00 MiB (GPU 0; 10.76 GiB total capacity; and also when I type nvidia-smi, the all the gpus are working! 

issue2: From the issue1, although 8 gpus mechine was runing train, but the speed is very slow, 50 batchs need over 8 mins. 
The same thing happend on 4 gpus mechine: 
command is ./tools/dist_train.sh configs/htc/htc_r50_fpn_1x_coco.py   4  --work-dir /home/work_dir_01/. 
The config of gpu_ids  still showed range(0, 1). 
The dataset is coco2017_val (only 5k images). cuda is NOT out of memery this time when using small dataset.
The speed of 50 batchs need over 5 mins.

So,
1. What's wroing with the config of  gpu_ids? why it always show  range(0, 1).
2. What's wrong with the speed of traing?  It seems like all gpus are cascaded but not parallelled. 

Please help me.
Thank you.

**Comment by ZwwWayne** [comment]
1. The `gpu_ids` might have some issues, but the training process should be fine.
2. As for the training speed, we are not sure because we've successfully run HTC many times with normal speeds. It depends on your GPU.

**Comment by tianxinhang** [comment]
Thank you. I'v succcessfully trained htc_r50_1e with 2 gpus. For the lr shedual, nothing has been changed but the lr rate decreased to 0.02/4. 
However, even in the last epoch (12th epoch). the loss is still around 0.8. Is this normal?

**Comment by ZwwWayne** [comment]
> Thank you. I'v succcessfully trained htc_r50_1e with 2 gpus. For the lr shedual, nothing has been changed but the lr rate decreased to 0.02/4.
> However, even in the last epoch (12th epoch). the loss is still around 0.8. Is this normal?

Yes it is normal because there are many losses in the cascade methods. You can compare the loss scale to those in the model zoo as we provide logs [here](https://github.com/open-mmlab/mmdetection/tree/master/configs/htc).
